File: tools/java/scenario-support/src/main/java/io/opentelemetry/conformance/scenario/ScenarioEnvironment.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.opentelemetry.conformance.scenario;

/** What the runner told a scenario, in the one place a scenario reads it. */
public final class ScenarioEnvironment {
  private ScenarioEnvironment() {}

  /** The value of {@code name}, or a failure naming what was missing. */
  public static String require(String name) {
    String value = System.getenv(name);
    if (value == null || value.isBlank()) {
      throw new IllegalStateException("required environment variable is missing: " + name);
    }
    return value;
  }

  /** The value of {@code name}, or {@code defaultValue} when it is missing or blank. */
  public static String getOrDefault(String name, String defaultValue) {
    String value = System.getenv(name);
    return value == null || value.isBlank() ? defaultValue : value;
  }
}

File: tools/dotnet/OpenTelemetry.Conformance.Scenario/ScenarioEnvironment.cs
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0

namespace OpenTelemetry.Conformance.Scenario;

/// <summary>What the runner told a scenario, in the one place a scenario reads it.</summary>
public static class ScenarioEnvironment
{
    /// <summary>The value of <paramref name="name"/>, or a failure naming what was missing.</summary>
    public static string Require(string name)
    {
        var value = Environment.GetEnvironmentVariable(name);
        if (string.IsNullOrWhiteSpace(value))
        {
            throw new InvalidOperationException(
                $"required environment variable is missing: {name}");
        }

        return value;
    }
}
